### PR TITLE
refactor: naming and interface

### PR DESCRIPTION
**Pull request by omermorad**

### CODE REVIEW

File: src/handlers/value-inspectors/primitive-value-inspector.ts [hunk 1]
```diff
@@ -29,14 +29,10 @@ export class PrimitiveValueInspector implements ValueInspector {
   }
 
   public deduceValue<T>(propertyDto: PropertyDto): any {
-    if (propertyDto.value) {
+    if (propertyDto.value !== 'undefined') {
```
Comment (idanpt-ni) [inline src/handlers/value-inspectors/primitive-value-inspector.ts:32]: remove

File: src/handlers/class-processor.ts [hunk 2]
```diff
@@ -1,26 +1,27 @@
 import reflect, { ClassReflection, PropertyReflection } from '@plumier/reflect';
-import { FunctionValueHandler } from './value-handlers/function-value-handler';
-import { ObjectLiteralValueHandler } from './value-handlers/object-literal-value-handler';
-import { EnumValueHandler } from './value-handlers/enum-value-handler';
-import { TypeValueInspector } from 'src/handlers/value-handlers/type-value-inspector';
-import { PrimitiveValueInspector } from 'src/handlers/value-handlers/primitive-value-inspector';
+import { FunctionValueInspector } from './value-inspectors/function-value-inspector';
+import { ObjectLiteralValueInspector } from './value-inspectors/object-literal-value-inspector';
+import { EnumValueInspector } from './value-inspectors/enum-value-inspector';
+import { TypeValueInspector } from './value-inspectors/type-value-inspector';
```
Comment (idanpt-ni) [inline src/handlers/class-processor.ts:5]: 1. 'Inspector' word is not accurate.
2. TypeValueInspector => ClassValueHandler

**Review by idanpt-ni — CHANGES_REQUESTED** [review]

File: src/handlers/primitive-handler-abstract.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import FakerStatic = Faker.FakerStatic;
+import { PropertyDto } from '../types/property-dto.interface';
+import { ExactValue } from '../types/exact-value.type';
+
+export abstract class PrimitiveHandlerAbstract {
+  protected static readonly PRIMITIVES = ['String', 'Boolean', 'Number', 'Date'];
```
Comment (idanpt-ni) [inline src/handlers/primitive-handler-abstract.ts:6]: extract this
PRIMITIVES = ['String', 'Boolean', 'Number', 'Date'];

to some outside const

File: src/class-processor.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+  public constructor(private readonly faker: FakerStatic, private readonly reflector: ClassReflector, locale: string) {
+    this.faker.setLocale(locale);
+  }
+
+  private handlePropertyValue(propertyDto: PropertyDto): T | T[] {
+    for (const inspectorClass of ClassProcessor.VALUE_INSPECTORS) {
+      const inspector = new inspectorClass(this.faker, this) as ValueInspector;
+
+      if (inspector.shouldInspect(propertyDto)) {
```
Comment (idanpt-ni) [inline src/class-processor.ts:35]: naming:

inspector => handler / producer
shouldInspect => shouldHandle 
deduceValue => handle / process / generateValue / makeValue / produceValue / ?

File: src/class-reflector.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
+import reflect, { ClassReflection, PropertyReflection } from '@plumier/reflect';
+import { ClassType } from './types/class.type';
+import { FixtureOptions } from './types/fixture-options.type';
+import { FIXTURE_DECORATOR_NAME } from './decorators/fixture.decorator';
+import { PropertyDto } from './types/property-dto.interface';
+import { ClassReflectionDto } from './types/class-reflection-dto.type';
+
+export class ClassReflector {
```
Comment (idanpt-ni) [inline src/class-reflector.ts:8]: needs own unit tests now :)

File: src/types/value-inspector.interface.ts [hunk 6]
```diff
@@ -1,11 +1,8 @@
-import { ClassReflection } from '@plumier/reflect';
 import { PropertyDto } from './property-dto.interface';
 import { IClassProcessor } from '../types/iclass-processor.interface';
 
 export interface ValueInspector {
   shouldInspect(propertyDto: PropertyDto): boolean;
 
-  hasCircularClassFixture(parentClassReflection: ClassReflection, propertyDto: PropertyDto): boolean;
```
Comment (idanpt-ni) [inline src/types/value-inspector.interface.ts:8]: reflect-metadata will not protect you from all circular references. you still need this check. please don't remove

File: src/handlers/value-inspectors/primitive-value-inspector.ts (hunk 1 above)
Comment (omer-morad-ni) [reply src/handlers/value-inspectors/primitive-value-inspector.ts]: We need it, it's very long to explain the case but I will show you on video

File: src/handlers/primitive-handler-abstract.ts (hunk 3 above)
Comment (omer-morad-ni) [reply src/handlers/primitive-handler-abstract.ts:6]: Why?

File: src/class-processor.ts (hunk 4 above)
Comment (omer-morad-ni) [reply src/class-processor.ts:35]: `produceValue` it is

File: src/handlers/primitive-handler-abstract.ts (hunk 3 above)
Comment (idanpt-ni) [reply src/handlers/primitive-handler-abstract.ts:7]: I saw 2 copies of this array in 2 files
  protected static readonly PRIMITIVES = ['String', 'Boolean', 'Number', 'Date'];

**Pull request closed without merging**